libraries/botframework-config/src/botConfiguration.ts

```diff
@@ -189,73 +265,168 @@ export class BotConfiguration extends BotConfigurationBase {
         }
     }
 
-    // save the config file to specificed botpath
-    public async saveAs(botpath: string, secret?: string): Promise<void> {
-        if (!botpath) {
-            throw new Error(`missing path`);
-        }
-
-        this._savePrep(secret);
-
-        const hasSecret: boolean = !!this.secretKey;
-
-        if (hasSecret) {
-            this.encrypt(secret);
-        }
-        await fsx.writeJson(botpath, this.toJSON(), { spaces: 4 });
-
-        if (hasSecret) {
-            this.decrypt(secret);
-        }
-    }
-
-    // save the config file to specificed botpath
-    public saveAsSync(botpath: string, secret?: string): void {
-        this._savePrep(secret);
+    // export the services from the bot file as resource files and recipe file
+    public async export(folder: string, progress?: (service: IConnectedService, command: string, index: number, total: number) => void): Promise<BotRecipe> {
```

Comment:
Consider a more clear way to write this, using sub-types or multiple lines of code. What's here is not clear for consumers of the code.